README.md

```diff
@@ -10,33 +10,103 @@ Telepoison is a [opentelemetry-instrumented](https://github.com/open-telemetry/o
+the `Keyword list` `opts` parameter (or the `t:HTTPoison.Request/0` `Keyword list` `options` parameter if calling `Telepoison.Request/1` directly). These are prefixed with `:ot_`.
+
+* `:ot_span_name` - sets the span name.
+* `:ot_attributes` - a list of `{name, value}` `tuple` attributes that will be added to the span.
+* `:ot_resource_route` - sets the `http.route` attribute, depending on the value provided.
+
+If the value is a string or an function with an arity of 1 (the `t:HTTPoison.Request/0` `request`) that is used to set the attribute
+
+If `:infer` is provided, then the function discussed within the [Configuration](#Configuration) section is used to set the attribute
+
+**It is highly recommended** to supply the `:ot_resource_route` explicitly as either a string or a function with an arity of 1 (the `t:HTTPoison.Request/0` `request`) 
+
+## Examples
+
+In the below examples, `Telepoison.get!/3` is used for the sake of simplicity but other functions derived from `Telepoison.request/1` can be used
+
+```elixir
+Telepoison.setup()
+
+Telepoison.get!(
+  "https://www.example.com/user/list",
+  [],
+  ot_span_name: "list example users",
+  ot_attributes: [{"example.language", "en"}],
+  ot_resource_route: :infer
+)
+```
+
+In the example above:
+* `Telepoison.setup/1` is called with no arguments
+* `:infer` is passed as the value for `:ot_resource_route` `Keyword list` option
 
-* `:ot_span_name` sets the span name
-* `:ot_attributes` additional span attributes that will be added to the span. Should be a list of {name, value} tuples.
+Given the above, the `http.route` attribute will be inferred as */user/:subpath*
 
+```elixir
+infer_fn = fn 
+  %HTTPoison.Request{} = request -> URI.parse(request.url).path
+end
+
+Telepoison.setup(infer_route: infer_fn)
+
+Telepoison.get!(
+  "https://www.example.com/user/list",
+  [],
+  ot_resource_route: :infer
+)
+```
+
+In the example above:
+* `Telepoison.setup/1` is called with the `:infer_route` `Keyword list` option set to a function which takes a `%HTTPoison.Request/0` argument, returning the path of the request URL
+* `:infer` is passed as the value for `:ot_resource_route` `Keyword list` option
+
+Given the above, the `http.route` attribute will be inferred as */user/list*
 
-Example:
 ```elixir
+Telepoison.setup()
+
 Telepoison.get!(
-  "en.wikipedia.org",
+  "https://www.example.com/user/list",
   [],
-  ot_span_name: "fetch wikipedia homepage",
-  ot_attributes: [{"wikipedia.language", "en"}]
+  ot_resource_route: "my secret path"
 )
 ```
 
+In the example above:
+* `Telepoison.setup/1` is called with the `:infer_route` `Keyword list` option set to a function which takes a `%HTTPoison.Request/0` argument, returning the path of the request URL
+* `"my secret path"` is passed as the value for `:ot_resource_route` `Keyword list` option
+
+Given the above, the `http.route` attribute will be inferred as *my secret path*
```

Comment:
I wouldn't say in this example the attribute is _inferred_ as `my secret path`, but it is set explicitly